esp-hal/src/usb_serial_jtag.rs

```diff
@@ -200,7 +200,7 @@ where
 
     /// Write data to the serial output in a non-blocking manner
     /// Requires manual flushing (automatically flushed every 64 bytes)
-    pub fn write_byte_nb(&mut self, word: u8) -> nb::Result<(), Error> {
+    pub fn write_byte_nb(&mut self, word: u8) -> Option<()> {
```

Comment:
Shall we keep `Option` result for the `_nb` methods? Otherwise they dont make much sense.

Reply:
Maybe if we give users a way to know if a function will block or not before calling it, we can get rid of the `_nb` functions completely. In this case it's easy but not sure if it's that easy everywhere